pkg/storage/buckets/store.go

```diff
@@ -126,17 +136,37 @@ func (b *BucketStore) List(ctx context.Context, startAfter, endBefore string, li
 		if itm.IsPrefix {
 			continue
 		}
+		m, err := convertMeta(itm.Meta)
+		if err != nil {
+			return items, more, err
+		}
 		items = append(items, ListItem{
 			Bucket: itm.Path,
-			Meta:   convertMeta(itm.Meta),
+			Meta:   m,
 		})
 	}
 	return items, more, nil
 }
 
 // convertMeta converts stream metadata to object metadata
-func convertMeta(m objects.Meta) Meta {
-	return Meta{
-		Created: m.Modified,
+func convertMeta(m objects.Meta) (Meta, error) {
+	var cipher storj.Cipher
+
+	pathEncType := m.UserDefined["path-enc-type"]
```

Comment:
Does it make sense to keep this in encrypted stream metadata? I suspect it would be more appropriate to have it configured per bucket rather than per stream?

Reply:
In the old way, buckets are stored as objects/streams with no content. Hence any bucket metadata end as stream metadata.

In the new metainfo way, we can do it much better.